param.go

```diff
@@ -287,13 +287,30 @@ func (po paramObject) DotParam() []*dot.Param {
 func newParamObject(t reflect.Type) (paramObject, error) {
 	po := paramObject{Type: t}
 
+	// Check if the In type supports unexported fields.
+	var allowUnexported bool
+	var err error
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		if f.Type == _inType {
+			allowUnexported, err = isUnexportedFieldAllowed(f)
```

Comment:
We can reduce the scope of the `err` to where it's first used:

```suggestion
	for i := 0; i < t.NumField(); i++ {
		f := t.Field(i)
		if f.Type == _inType {
			var err error
			allowUnexported, err = isUnexportedFieldAllowed(f)
```